lib/internal/errors.js

```diff
@@ -111,7 +111,23 @@ module.exports = exports = {
 //
 // Note: Please try to keep these in alphabetical order
 E('ERR_ARG_NOT_ITERABLE', '%s must be iterable');
+E('ERR_ARG_OUT_OF_BOUNDS',
+  (name) => `The "${name}" argument is out of bounds`);
 E('ERR_ASSERTION', (msg) => msg);
+E('ERR_BUFFER_ENCODING_PROVIDED',
+  'If encoding is specified then the first argument must be a string');
+E('ERR_BUFFER_INVALID_CONCAT_LIST',
+  'The "list" argument must be an Array of Buffer or Uint8Array instances');
```

Comment:
There's the whole over specialization of invalid options debate:
```js
try {
  // do some stuff
  var f = fs.openSync('g.txt', 'rgh');
  // do other stuff
catch (e) {
  if (e.code == 'ERR_INVALID_OPT_VALUE') console.error(e) && process.exit(1) // because it's a programming error
  else ...;
}
```
Maybe worth creating a new type? `OptionError`, or have all the codes include a substring?